src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -720,20 +720,29 @@ TerminalInput::OutputType Terminal::SendCharEvent(const wchar_t ch, const WORD s
     // Then treat this line like it's a prompt mark.
     if (_autoMarkPrompts && vkey == VK_RETURN && !_inAltBuffer())
     {
-        // * If we have a current prompt:
-        //   - Then we did know that the prompt started, (we may have also
-        //     already gotten a MarkCommandStart sequence). The user has pressed
-        //     enter, and we're treating that like the prompt has now ended.
-        //     - Perform a FTCS_COMMAND_EXECUTED, so that we start marking this
-        //       as output.
-        //     - This enables CMD to have full FTCS support, even though there's
-        //       no point in CMD to insert a "pre exec" hook
-        // * Else: We don't have a prompt. We don't know anything else, but we
-        //   can set the whole line as the prompt, no command, and start the
-        //   command_executed now.
+        // We need to be a little tricky here, to try and support folks that are
+        // auto-marking prompts, but don't necessarily have the rest of shell
+        // integration enabled.
         //
-        // Fortunately, MarkOutputStart will do all this logic for us!
-        MarkOutputStart();
+        // We'll set the current attributes to Output, so that the output after

```

Comment:
Ah yea, this one's the wackiest case of them all. In this particular case, we want to support folks that are using `autoMarkPrompts` all by its lonesome, without any other shell integration. So yea, we're doing some time travelling here, so that their buffer kinda looks like they had shell integration enabled the whole time. That way they can at least jump between prompts with `scrollToMark`, but `selectCommand` won't quite work as expected.